src/main/java/bio/terra/stairway/Stairway.java

```diff
@@ -258,20 +279,37 @@ public Stairway(Stairway.Builder builder) {
             ? "stairway" + UUID.randomUUID().toString()
             : builder.stairwayName;
 
-    this.stairwayClusterName =
+    String stairwayClusterName =
         (builder.stairwayClusterName == null)
             ? "stairwaycluster" + UUID.randomUUID().toString()
             : builder.stairwayClusterName;
 
-    this.classLoader =
-        (builder.classLoader == null) ? this.getClass().getClassLoader() : builder.classLoader;
-
     this.flightFactory =
         (builder.flightFactory == null) ? new StairwayFlightFactory() : builder.flightFactory;
 
+    this.workQueueProjectId = builder.workQueueProjectId;
+    this.workQueueEnabled = (workQueueProjectId != null && builder.enableWorkQueue);
+    if (!workQueueEnabled) {
+      workQueueCreate = false;
+      this.workQueueTopicId = null;
+      this.workQueueSubscriptionId = null;
+    } else {
+      if (builder.workQueueTopicId == null) {
+        workQueueCreate = true;
```

Comment:
Just sanity checking that this is the desired behavior: Your above comment [line 194] says that if the workQueueProjectId is not supplied, no queue will be used. This set of logics appears to allow enableWorkQueue override this - if enableWorkQueue is true, then workQueueProjectId could be null and we would still use the work queue.

Reply:
`this.workQeueEnabled` is only true if `enableWorkQueue` is true and `builder.workQueueProjectId` is not null.
I think this is ok